src/query.rs

```diff
@@ -183,8 +190,42 @@ pub struct QueryOverride {
     pub lint_level: Option<LintLevel>,
 }
 
-/// A mapping of lint ids to configured values that override that lint's defaults.
-pub type OverrideMap = BTreeMap<String, QueryOverride>;
+/// A mapping of `Identifier`s to configured values that override those lints' defaults.
+pub type OverrideMap = BTreeMap<Identifier, QueryOverride>;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum LintGroup {
+    MustUseAdded,
+}
+
+/// An identifier that matches a lint, or a group of lints.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Identifier {
+    /// Identifies a group of lints, based on the [`SemverQuery::lint_group`] field.
+    Group(LintGroup),
+    /// Identifies a singular lint by id.
+    Lint(String),
+}
+
+impl Identifier {
+    /// Returns a list of lint ids that this identifier refers to, given the set of
+    /// all queries.
+    pub(crate) fn lints<'a>(
```

Comment:
If/when we add dynamic groups like `warnings`, `major`, and `minor`, this will need an extra `&OverrideStack` argument to determine this.  We call this method in `OverrideStack::try_push`, so this will be an easy change.